### PR TITLE
ColumnUserFilter: extract text format logic in method to allow override

### PR DESCRIPTION
**Pull request by nsteger**

### CODE REVIEW

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      else {
+        t = days + " " + TEXTS.get("Day") + " ";
+      }
+    }
+    t = t + StringUtility.lpad("" + h, "0", 2) + ":" + StringUtility.lpad("" + m, "0", 2) + ":" + StringUtility.lpad("" + s, "0", 2);
+    return t;
+  }
+
+  /**
+   * Format: "d hh:mm:ss" = d [day(s)] hh [hour] : mm [minute] : ss [second] . zzz [millisecond]
```
Comment (egloff) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java]:
```suggestion
   * Format: "d hh:mm:ss.zzz" = d [day(s)] hh [hour] : mm [minute] : ss [second] . zzz [millisecond]
```

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+   * </ul>
+   * * Example: {@code "1 day 15:42:54.002"}
+   */
+  public String formatTimePeriodOfMs(Long milliSecs) {
+    char decimalSeparator = BEANS.get(NumberFormatProvider.class).getNumberInstance(NlsLocale.get()).getDecimalFormatSymbols().getDecimalSeparator();
+    if (milliSecs == null || milliSecs.longValue() <= 0) {
+      return "00:00:00" + decimalSeparator + "000";
+    }
+
+    BigDecimal msConvertedToDays = new BigDecimal(milliSecs - (milliSecs % 1000)).divide(new BigDecimal(1000 * 60 * 60 * 24), MathContext.DECIMAL128);
```
Comment (egloff) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java]: `60 * 60 * 24 = 86400` is used in formatTimePeriod as well, may be we could introduce a constant and multiply by 1000 here...

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+   * <ul>
+   * <li>1 day = 1.0d</li> *
+   * <li>If the argument d is negative, the result is '00:00:00.000'</li>
+   * <li>Fraction of seconds are rounded with {@link RoundingMode#HALF_UP}</li>
+   * </ul>
+   * * Example: {@code "1 day 15:42:54.002"}
+   */
+  public String formatTimePeriodOfMs(Long milliSecs) {
+    char decimalSeparator = BEANS.get(NumberFormatProvider.class).getNumberInstance(NlsLocale.get()).getDecimalFormatSymbols().getDecimalSeparator();
+    if (milliSecs == null || milliSecs.longValue() <= 0) {
```
Comment (egloff) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java]: Shouldn't this return null if milliSecs is null?

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    String secondsPattern = secondSymbol + secondSymbol;
+
+    String millisecondSymbol = Character.toString(localPatternChars.charAt(DateFormat.MILLISECOND_FIELD));
+    String millisecondsPattern = millisecondSymbol + millisecondSymbol + millisecondSymbol;
+
+    // Add milliseconds with decimalSeparator after seconds.
+    DecimalFormat decimalFormat = BEANS.get(NumberFormatProvider.class).getNumberInstance(NlsLocale.get());
+    char decimalSeparator = decimalFormat.getDecimalFormatSymbols().getDecimalSeparator();
+    DateFormat dateFormatWithSeconds = BEANS.get(DateFormatProvider.class).getDateTimeInstance(DateFormat.SHORT, DateFormat.MEDIUM, NlsLocale.get());
+    String pattern = ((SimpleDateFormat) dateFormatWithSeconds).toPattern();
```
Comment (nsteger) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/date/DateTimePeriodFormatter.java]: instanceof

**Review by egloff — APPROVED** [review]

**Pull request merged**